src/device/index.ts

```diff
@@ -0,0 +1,55 @@
+//
+// Playpass (c) Playco
+// https://github.com/playpassgames/playpass/blob/main/LICENSE.txt
+
+import { ShareType } from "../share/share-type";
+
+type Detector = {
+    shareType: ShareType;
+
+    // If either one of these tests pass, it's a hit
+    userAgent?: RegExp;
+    referrer?: RegExp;
+};
+
+const detectors: Detector[] = [
+    {
+        shareType: ShareType.Facebook,
+
+        // https://github.com/paypal/paypal-checkout-components/issues/484
+        // https://gist.github.com/taylorhughes/dae564f63b3d702429a5f37bd22b97b8
+        userAgent: /\b(FBAV|FBAN|FB_IAB)\//,
+
+        // l.facebook.com and lm.facebook.com
+        referrer: /\bfacebook\.com$/,
+    },
+    {
+        shareType: ShareType.Twitter,
+
+        // No userAgent detection available on Twitter!
+
+        // twitter.com seems unused, but included here for safety
+        referrer: /\b(t\.co|twitter\.com)$/,
+    },
+
+    // TODO(2022-06-23): Add Reddit?
```

Comment:
yes pls

Reply:
reddit referrer should be `/b(redd.it|reddit\.com)$/` as far as I can tell